### PR TITLE
Article URLs

### PR DESCRIPTION
**Pull request by kallehauge**
This PR makes it possible to organise articles in a "tree like" URL structure and adds the possibility to customise the slug.

### CODE REVIEW

File: web/config/sync/core.extension.yml [hunk 1]
```diff
@@ -22,15 +22,18 @@ module:
...
   serialization: 0
   shortcut: 0
   system: 0
   taxonomy: 0
   text: 0
+  token: 0
   toolbar: 0
   user: 0
   video_embed_field: 0
+  pathauto: 1
```
Comment (kasperg) [inline web/config/sync/core.extension.yml:36]: Why does this module have the value `1` while the others have `0`?

File: web/modules/custom/dbcdk_community_content/dbcdk_community_content.module [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    $node->set('field_slug', $alias_cleaner->cleanString($node->getTitle()));
+  }
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Make the "Slug" and "Sub Path" fields part of an advanced details group.
+ * We do this to give a better editorial experience by placing all additional
+ * options for the content together.
```
Comment (kasperg) [inline web/modules/custom/dbcdk_community_content/dbcdk_community_content.module:31]: :+1:

File: web/modules/custom/dbcdk_community_content/dbcdk_community_content.tokens.inc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * referenced entities "sub values".
+ */
+function dbcdk_community_content_tokens($type, array $tokens, array $data, array $options, BubbleableMetadata $bubbleable_metadata) {
+  $replacements = array();
+  if ($type === 'entity' & !empty($data['entity'])) {
+    /** @var \Drupal\Core\Entity\EntityInterface $entity */
+    $entity = $data['entity'];
+    foreach ($tokens as $name => $original) {
+      switch ($name) {
+        case 'sub-path':
```
Comment (kasperg) [inline web/modules/custom/dbcdk_community_content/dbcdk_community_content.tokens.inc]: How about replacing `foreach/switch/case` with an `if ($tokens['sub-path'])`?

That seems a bit simpler than looping over all tokens.

File: web/modules/custom/dbcdk_community_content/dbcdk_community_content.module [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+/**
+ * Implements hook_node_presave().
+ *
+ * Duplicate the title of a node to the slug field if the slug field is empty.
+ * We do this because article urls depend on having a slug field value for the
+ * article Pathauto pattern.
+ */
+function dbcdk_community_content_node_presave(Node $node) {
+  if (empty($node->get('field_slug')->getString())) {
+    /** @var \Drupal\pathauto\AliasCleaner $alias_cleaner */
```
Comment (kasperg) [inline web/modules/custom/dbcdk_community_content/dbcdk_community_content.module]: Nitpicking: Double `**` here.

**Comment by kasperg** [comment]
In general I think this looks good. I have a few minor issues here and there.

One thing: We talked about the ability to filter content based on subpath. Will that be added in a later PR? I could not spot an issue for it.

File: web/config/sync/core.extension.yml (hunk 1 above)
Comment (kasperg) [inline web/config/sync/core.extension.yml:36]: It is the module weight. Never mind :+1:.

**Comment by kallehauge** [comment]
Corrected the issues @kasperg  :relaxed:

Regarding "filter content based on subpath", are you referring to the list of content? If so, then it's a separate ticket that haven't been handlede yet.

**Pull request merged**